projects/igniteui-angular/src/lib/icon/icon.component.ts

```diff
@@ -32,63 +32,190 @@ export class IgxIconComponent implements OnInit {
     @ViewChild('explicitLigature', { read: TemplateRef })
     private explicitLigature: TemplateRef<HTMLElement>;
 
+    /**
+     *  This allows you to disable `igx-icon` class. By default it's applied.
+     *```html
+     *@ViewChild("MyIcon")
+     *public icon: IgxIconComponent;
+     *ngAfterViewInit() {
+     *    icon.cssClass = false;
```

Comment:
cssClass is not a flag, description should be changed. Also it won't allow you set it to false, it should be a string. And use change detector to avoid errors on the page.
    @ViewChild("MyIcon") public icon: IgxIconComponent;

    constructor(private cdRef:ChangeDetectorRef) {}

    ngAfterViewInit() {
        this.icon.cssClass = "";
        this.cdRef.detectChanges();
    }